src/libraries/System.Security.Cryptography.ProtectedData/src/System/Security/Cryptography/ProtectedData.cs

```diff
@@ -19,7 +22,148 @@ public static byte[] Protect(byte[] userData, byte[]? optionalEntropy, DataProte
             if (userData is null)
                 throw new ArgumentNullException(nameof(userData));
 
-            return ProtectOrUnprotect(userData, optionalEntropy, scope, protect: true);
+            byte[]? outputData;
+            bool result = TryProtectOrUnprotect(
+                userData,
+                optionalEntropy,
+                scope,
+                protect: true,
+                allocateArray: true,
+                bytesWritten: out _,
+                outputData: out outputData
+            );
+            Debug.Assert(result);
+            Debug.Assert(outputData != null);
+            return outputData;
+        }
+
+        /// <summary>
+        /// Encrypts the data in a specified byte span and returns a byte array that contains the encrypted data.
+        /// </summary>
+        /// <param name="userData">A byte span that contains data to encrypt.</param>
+        /// <param name="scope">One of the enumeration values that specifies the scope of encryption.</param>
+        /// <param name="optionalEntropy">
+        /// An optional additional byte span used to increase the complexity of the encryption,
+        /// or empty for no additional complexity.
+        /// </param>
+        /// <returns>A byte array representing the encrypted data.</returns>
+        /// <exception cref="CryptographicException">The encryption failed.</exception>
+        /// <exception cref="NotSupportedException">The operating system does not support this method.</exception>
+        /// <exception cref="OutOfMemoryException">The system ran out of memory while encrypting the data.</exception>
+        /// <exception cref="PlatformNotSupportedException">
+        /// Calls to the Protect method are supported on Windows operating systems only.
+        /// </exception>
+        public static byte[] Protect(
+            ReadOnlySpan<byte> userData,
+            DataProtectionScope scope,
```

Comment:
This is a problem with the existing methods, but we don't validate that the `scope` is a valid enum value:

```C#
ProtectedData.Protect(new byte[3], null, (DataProtectionScope)54);
```

On one hand, if we fix it, then it will be divergent behavior from .NET Framework. We could only fix it for "new" methods, but that will be weird.

Reply:
Nevermind, that is "documented"

> // For .NET Framework compat, we ignore unknown bits in the "scope" value rather than throwing.

Bleh.